.ci/Jenkinsfile

```
@@ -102,25 +110,70 @@ pipeline {
                 unsuccessful {
                   archiveArtifacts(allowEmptyArchive: true, artifacts: "${MSBUILDDEBUGPATH}/**/MSBuild_*.failure.txt")
                 }
+                always {
+                  dir("${BASE_DIR}"){
+                    junit(allowEmptyResults: true, keepLongStdio: true, testResults: 'build/output/junit-*.xml')
+                    archiveArtifacts(allowEmptyArchive: true, artifacts: 'build/output/**/*')
+                  }
+                }
               }
             }
-            stage('Deploy') {
-              when {
-                branch 'master'
-              }
-              steps {
-                dir("${BASE_DIR}"){
-                  release('secret/apm-team/ci/elastic-observability-appveyor')
-                }
+          }
+        }
+      }
+    }
+    stage('Release') {
+      when {
+        beforeAgent true
+        anyOf {
+          tag pattern: '\\d+\\.\\d+\\.\\d+(-(alpha|beta|rc)\\d*)?', comparator: 'REGEXP'
+          branch 'master'
+        }
+      }
+      stages {
+        stage('Generate') {
+          steps {
+            deleteDir()
+            unstash 'source'
+            dir("${BASE_DIR}"){
+              dotnet() {
+                sh(label: 'Generate local nuget packages', script: './build.sh generatepackages -s true')
+                sh(label: 'Validate *.npkg files that were created', script: './build.sh validatepackages -s true')
+                sh(label: 'Inspect public API change', script: './build.sh generateapichanges -s true')
               }
-              post {
-                always {
-                  archiveArtifacts(allowEmptyArchive: true, artifacts: "${BASE_DIR}/**/build/output/*.nupkg")
+            }
+          }
+        }
+        stage('Release to feedz.io') {
+          when { branch 'master' }
+          steps {
+            dir("${BASE_DIR}"){
+              deploy(secret: 'secret/apm-team/ci/elastic-observability-feedz.io')
+            }
+          }
+        }
+        stage('Release Notes') {
+          when { buildingTag() }
+          steps {
+            dir("${BASE_DIR}"){
+              withGitRelease(credentialsId: '2a9602aa-ab9f-4e52-baf3-b71ca88469c7-UserAndToken') {
+                dotnet() {
+                  sh(label: 'Generate release notes', script: './build.sh generatereleasenotes -s true')
+                  sh(label: 'Inspect public API change', script: './build.sh generateapichanges -s true')
+                  sh(label: 'Create or update release for tag on github', script: "./build.sh createreleaseongithub -s true --token ${GITHUB_TOKEN}")
                 }
               }
             }
           }
         }
+        stage('Release to NuGet') {
+          when { buildingTag() }
+          steps {
+            dir("${BASE_DIR}"){
+              //deploy(secret: 'secret/apm-team/ci/elastic-observability-nuget')
```

@Mpdreamz , is that ok?
```suggestion
              deploy(secret: 'secret/apm-team/ci/elastic-observability-nuget')
```